datastew/repository/weaviate.py

```diff
@@ -17,15 +19,30 @@
 class WeaviateRepository(BaseRepository):
     logger = logging.getLogger(__name__)
 
-    def __init__(self, mode="memory", path=None, port=80):
+    def __init__(self, mode="memory", path=None, port=80, http_port=8079, grpc_port=50050):
         self.mode = mode
+        self.client: Union[None, WeaviateClient] = None
         try:
             if mode == "memory":
-                self.client = weaviate.connect_to_embedded(persistence_data_path="db")
+                # Check if there is an existing instance of Weaviate client for the default ports
+                if self._is_port_in_use(http_port) and self._is_port_in_use(grpc_port):
+                    # if the client exists, first close then re-connect
+                    if self.client:
+                        self.client.close()
+                    self.client = weaviate.connect_to_local(port=http_port, grpc_port=grpc_port)
```

Comment:
I think this is still embedded. This only connects to the local if this specific port is already used (by the Embedded). For example, in the case of a small disconnection, the port remains occupied, but the local instance is lost.

However, I'm not 100% sure because their documentation does not cover this. Only [the error message](https://github.com/SCAI-BIO/datastew/actions/runs/11660827244/job/32463937467) suggested re-connecting to this specific port because a client is already in use.